napari/types.py

```diff
@@ -1,20 +1,13 @@
 from functools import wraps
-from typing import Any, Callable, Dict, List, Protocol, Tuple, Union
+from typing import Any, Callable, Dict, List, Tuple, Union, NewType
 
 import numpy as np
 
-
-class ArrayLike(Protocol):
-    """This is a WOEFULLY inadqueate stub for a duck-array type.
-
-    Mostly, just a placeholder for the concept of needing an ArrayLike type.
-    Ultimately, this should come from https://github.com/napari/image-types
-    """
-
-    shape: Tuple[int, ...]
-    ndim: int
-    dtype: np.dtype
-
+# This is a WOEFULLY inadqueate stub for a duck-array type.
+# Mostly, just a placeholder for the concept of needing an ArrayLike type.
+# It doesn't actually get used for type checking anywhere.
+# Ultimately, this should come from https://github.com/napari/image-types
+ArrayLike = NewType("ArrayLike", np.array)
```

Comment:
Where does it get used if not in type checking? This is even more inadequate than before. =P

Reply:
i was at a loss :). what should I use then for `ArrayLike` in the type annotation in 
```python
def image_reader_to_layerdata_reader(
    func: Callable[[PathLike], ArrayLike]
) -> ReaderFunction:
```

Reply:
Well, it should at least be a `Union[np.array, dask.Array, zarr.Array]` (or whatever those types actually are, I haven't checked. =)

Reply:
Also, there should be a note that this should be a Protocol once we move to 3.7, is that right?

Reply:
> Well, it should at least be a Union[np.array, dask.Array, zarr.Array]

I skimped out on that because I would have to `try: import zarr` since we don't depend on it.  But I can do that

> there should be a note that this should be a Protocol once we move to 3.7, is that right?

`Protocol` is actually 3.8, but can be used in ≥3.5 if we depend on the [typing-extensions](https://github.com/python/typing/tree/master/typing_extensions) backport.